tests/artifacts/test_artifacts.py

```diff
@@ -344,6 +344,22 @@ def test_log_artifact_with_target_path_and_upload_options():
                 assert logged_artifact.metadata.hash is None
 
 
+@pytest.mark.parametrize(
+    "local_path, fail", [("s3://path/file.txt", False), ("/not_exists/file.txt", True)]
```

Comment:
Since you did advanced validation on local paths, might as well check it here... Add some test cases for things like `file://some/path/file.txt` or `C:/hello/world.bat`.